src/util/type_eq.cpp

```diff
@@ -1,20 +1,26 @@
 /*******************************************************************\
 
-Module: Type Checking
+Module: Type checking
 
 Author: Daniel Kroening, kroening@kroening.com
+        Maria Svorenova, maria.svorenova@diffblue.com
 
 \*******************************************************************/
 
 /// \file
-/// Type Checking
+/// Type checking
 
 #include "type_eq.h"
 
 #include "namespace.h"
 #include "std_types.h"
 #include "symbol.h"
 
+/// Check types for equality. If either of the types is a symbol type, i.e. a
```

Comment:
I have never quite understood when to use `type_eq` vs `base_type_eq`. I would appreciate if the documentation introduced could please sort this out.